packages/notebook/src/truststatus.tsx

```diff
@@ -205,11 +239,64 @@ export namespace NotebookTrustStatus {
 
         this._totalCells = total;
         this._trustedCells = trusted;
+        newNotebook.disposed.connect(() => {
+          if (newNotebook.model) {
+            newNotebook.model.cells.changed.disconnect(this.trustCell);
+          }
+        });
       }
 
       this._triggerChange(oldState, this._getAllState());
     }
 
+    /**
+     * When the cell list is changed, trust the added cell.
+     */
+    trustCell(
+      _: IObservableUndoableList<ICellModel>,
+      args: IObservableList.IChangedArgs<ICellModel>
+    ): void {
+      if (args.type === 'add') {
+        args.newValues.forEach(cell => {
+          cell.trusted = true;
+        });
+      }
+    }
+
+    /**
+     * Toggle the trust status of all cells in notebook
+     */
+    toggleTrustNotebook(trust: boolean): void {
+      if (this._notebook) {
+        const { context, content } = this._notebook;
+        if (!content.model) {
+          return;
+        }
+        const cells = toArray(content.model.cells);
+        cells.forEach(cell => (cell.trusted = trust));
+        context.save();
```

Comment:
Could you confirm that this is the reason you need NotebookPanel instead of Notebook?